server/api/projectfiles.ts

```diff
@@ -1132,60 +1111,115 @@ async function removeProjectFile(
       })
       .filter((obj) => obj !== null);
 
-    if (filesToDelete.length > 0) {
-      const storageClient = new S3Client(PROJECT_FILES_S3_CLIENT_CONFIG);
-      const deleteRes = await storageClient.send(
+  if (filesToDelete.length > 0) {
+    const storageClient = new S3Client(PROJECT_FILES_S3_CLIENT_CONFIG);
+    const deleteRes = await storageClient.send(
         new DeleteObjectsCommand({
           Bucket: process.env.AWS_PROJECTFILES_BUCKET,
           Delete: {
             Objects: filesToDelete as { Key: string }[],
           },
         })
-      );
-      if (Array.isArray(deleteRes.Errors) && deleteRes.Errors.length > 0) {
-        return res.status(500).send({
-          err: true,
-          errMsg: conductorErrors.err58,
-        });
-      }
+    );
+    if (Array.isArray(deleteRes.Errors) && deleteRes.Errors.length > 0) {
+      throw new Error('delete_errors_encountered');
     }
+  }
 
-    if (videosToDelete.length > 0) {
-      const deletePromises = videosToDelete.map((videoID) => {
-        const ENDPOINT = `https://api.cloudflare.com/client/v4/accounts/${process.env.CLOUDFLARE_STREAM_ACCOUNT_ID}/stream/${videoID}`;
-        return axios.delete(ENDPOINT, {
-          headers: {
-            Authorization: `Bearer ${process.env.CLOUDFLARE_STREAM_API_TOKEN}`,
-          },
-        });
+  if (videosToDelete.length > 0) {
+    const deletePromises = videosToDelete.map((videoID) => {
+      const ENDPOINT = `https://api.cloudflare.com/client/v4/accounts/${process.env.CLOUDFLARE_STREAM_ACCOUNT_ID}/stream/${videoID}`;
+      return axios.delete(ENDPOINT, {
+        headers: {
+          Authorization: `Bearer ${process.env.CLOUDFLARE_STREAM_API_TOKEN}`,
+        },
       });
+    });
 
-      await Promise.all(deletePromises);
-    }
+    await Promise.all(deletePromises);
+  }
 
-    const deleteRes = await ProjectFile.deleteMany({
-      projectID,
-      fileID: { $in: objectIDs },
-    });
+  if (!fileIDs?.length) {
+    return;
+  }
+  await ProjectFile.deleteMany({
+    projectID,
+    fileID: { $in: fileIDs },
+  });
+}
+
+/**
+ * Deletes a Project File from the database and underlying storage solution. Multiple files can be deleted by
+ * specifying a folder identifier.
+ *
+ * @param projectID - Project identifier.
+ * @param fileID - Project File (or folder) identifier.
+ */
+async function _removeProjectFile(projectID: string, fileID: string) {
```

Comment:
why create another function for plurality when we could just pass the id as lone array element?